WordPress/Sniffs/WhiteSpace/ArbitraryParenthesesSpacingSniff.php

```diff
@@ -0,0 +1,256 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPress\Sniffs\WhiteSpace;
+
+use WordPress\Sniff;
+use PHP_CodeSniffer_Tokens as Tokens;
+
+/**
+ * Check & fix whitespace on the inside of arbitraty parentheses.
+ *
+ * Arbitrary parenthesis are those which are not owned by a function (call), array or control structure.
+ * Spacing on the outside is not checked on purpose as this would too easily conflict with other spacing rules.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.14.0
+ *
+ * {@internal This sniff is a duplicate of the same sniff as pulled upstream.
+ * Once the upstream sniff has been merged and the minimum WPCS PHPCS requirement has gone up to
+ * the version in which the sniff was merged, this version can be safely removed.
+ * {@link https://github.com/squizlabs/PHP_CodeSniffer/pull/1701} }}
+ */
+class ArbitraryParenthesesSpacingSniff extends Sniff {
+
+	/**
+	 * The number of spaces desired on the inside of the parentheses.
+	 *
+	 * @since 0.14.0
+	 *
+	 * @var integer
+	 */
+	public $spacingInside = 0;
+
+	/**
+	 * Allow newlines instead of spaces.
+	 *
+	 * @since 0.14.0
+	 *
+	 * @var boolean
+	 */
+	public $ignoreNewlines = false;
+
+	/**
+	 * Tokens which when they precede an open parenthesis indicate
+	 * that this is a type of structure this sniff should ignore.
+	 *
+	 * @since 0.14.0
+	 *
+	 * @var array
+	 */
+	private $ignoreTokens = array();
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @since 0.14.0
+	 *
+	 * @return array
+	 */
+	public function register() {
+
+		$this->ignoreTokens                           = Tokens::$functionNameTokens;
+		$this->ignoreTokens[ T_VARIABLE ]             = T_VARIABLE;
+		$this->ignoreTokens[ T_CLOSE_PARENTHESIS ]    = T_CLOSE_PARENTHESIS;
+		$this->ignoreTokens[ T_CLOSE_CURLY_BRACKET ]  = T_CLOSE_CURLY_BRACKET;
+		$this->ignoreTokens[ T_CLOSE_SQUARE_BRACKET ] = T_CLOSE_SQUARE_BRACKET;
+		$this->ignoreTokens[ T_CLOSE_SHORT_ARRAY ]    = T_CLOSE_SHORT_ARRAY;
+		$this->ignoreTokens[ T_ANON_CLASS ]           = T_ANON_CLASS;
+		$this->ignoreTokens[ T_USE ]                  = T_USE;
+		$this->ignoreTokens[ T_LIST ]                 = T_LIST;
+		$this->ignoreTokens[ T_DECLARE ]              = T_DECLARE;
+
+		// The below two tokens have been added to the Tokens::$functionNameTokens array in PHPCS 3.1.0,
+		// so they can be removed once the minimum PHPCS requirement of WPCS has gone up.
+		$this->ignoreTokens[ T_SELF ]   = T_SELF;
+		$this->ignoreTokens[ T_STATIC ] = T_STATIC;
+
+		// Language constructs where the use of parentheses should be discouraged instead.
```

Comment:
Does require / require_once / include / include_once need to be included here as well?

Reply:
They already are via `Tokens::$functionNameTokens`.

See: https://github.com/squizlabs/PHP_CodeSniffer/blob/master/src/Util/Tokens.php#L521-L534